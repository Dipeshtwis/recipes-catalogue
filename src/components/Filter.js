import React from 'react';
import PropTypes from 'prop-types';

const Filter = ({ handleFilter, categories }) => (
  <>
    <h2>Categories</h2>
    <div className="category">
      {
        categories.map(category => (
          <div key={category.idCategory} className="category-item">
            <img src={category.strCategoryThumb} alt="{category.strCategory}" className="category-img" />
            <button
              type="button"
              key={category.idCategory}
              onClick={handleFilter}
              onKeyDown={handleFilter}
              className="category-button"
            >
              {category.strCategory}
            </button>
          </div>
        ))
      }
    </div>
  </>
);

Filter.propTypes = {
  handleFilter: PropTypes.func.isRequired,
  categories: PropTypes.oneOfType([PropTypes.array, PropTypes.object]).isRequired,
};

export default Filter;
