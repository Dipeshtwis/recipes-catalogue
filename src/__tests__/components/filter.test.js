import React from 'react';
import renderer from 'react-test-renderer';
import Filter from '../../components/Filter';

it('renders filter component correctly', () => {
  const filter = renderer.create(<filter />).toJSON();
  expect(filter).toMatchSnapshot();
});