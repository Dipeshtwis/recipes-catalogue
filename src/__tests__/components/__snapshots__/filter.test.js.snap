// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `<filter />`;

exports[`renders filter component correctly 1`] = `<filter />`;
